docs/markdoc-component-documentation.md

```diff
@@ -0,0 +1,263 @@
+        "Yes – definitely.",
+        "You may rely on it.",
+        "Absolutely not!",
+        "Go for it!",
+        "No way, José!",
+        "Oh, hell no!",
+        "Hell yes!",
+        "Yes, if you believe hard enough.",
+        "No, but that shouldn't stop you.",
+        "Why would you even ask that?"
+    ]
+
+    while True:
+        question = input("Ask the Magic 8 Ball a question (type 'exit' to quit): ")
+        if question.lower() == 'exit':
+            print("Magic 8 Ball has left the chat.")
+            break
+        else:
+            print(random.choice(responses))
+
+if __name__ == "__main__":
+    magic_8_ball()
+
+```
+
+# Custom Markdoc Tags
+We've extended markdown with custom tags that let us make the docs richer and more interactive.
+
+## Admonitions
+
+Two types of admonitions are available. Warnings and notes.
+
+`{% warning %} This is a warning {% /warning %}`
+
+{% warning %} This is a warning {% /warning %}
+
+`{% note %} This is a note {% /note %}`
+
+{% note %} This is a note {% /note %}
+
+## Buttons
+Buttons are basically just links that are styled to look neat to the user and work for making links stand out.
+
+We have a few different styles of buttons that all do the same thing.
+
+### Default
+
+`{% button link="https://dog.ceo/" %} Click Me! {% /button %}`
+
+{% button link="https://dog.ceo/" %} Click Me! {% /button %}
+
+### Primary
+
+`{% button link="https://dog.ceo/" style="primary" %} Click Me! {% /button %}`
+
+{% button link="https://dog.ceo/" style="primary" %} Click Me! {% /button %}
+
+### Secondary
+
+`{% button link="https://dog.ceo/" style="secondary" %} Click Me! {% /button %}`
+
+{% button link="https://dog.ceo/" style="secondary" %} Click Me! {% /button %}
+
+### Blurple
+
+` {% button link="https://dog.ceo/" style="blurple" %} Click Me! {% /button %} `
+
+{% button link="https://dog.ceo/" style="blurple" %} Click Me! {% /button %}
+
+### Button Container
+The main use case I've seen for buttons in the docs is basically setting up multiple styling links in a row. Doing this requires putting buttons into a `ButtonContainer`
+
+{% buttonContainer %} {% button link="https://dog.ceo/" style="primary" %} Click Me! {% /button %} {% button link="https://dog.ceo/" style="secondary" %} Click Me! {% /button %} {% button link="https://dog.ceo/" style="blurple" %} Click Me! {% /button %} {% /buttonContainer %}
+
+## Crosses and Checks 
+You can invoke a cross with this tag `{% cross /}%` and it looks like this {% cross /%}. You can invoke checks with this tag `{% check /}%` and it looks like this {% check /%}.
+
+Crosses and checks can also be used in lists:
+- {% check /%} Completed task
+- {% cross /%} Incomplete task
+- {% cross /%} Super Incomplete task 
+
+You can also put crosses and checks into headers 
+
+### This is a header with a check {% check /%}
+Which is pretty neat.
+
+### This is a header with a cross {% cross /%}
+Which is also pretty neat.
+
+## Images
```

Comment:
When using this, do we have to specify the dimensions? Currently, if we use Markdown images and then run `make mdx-format`, the script will convert the images to `<Image>` tags and add the dimensions to the tag for us. It'd be nice to not have to do this manually - I can see myself and others forgetting to do it.

Reply:
I did not know that script existed. That is a much more convenient way to approach that. Let me research how it works and see if I can't either extend it to work on Markdoc or make a new script for the Markdoc tags.

Reply:
🙏